flink-test-utils-parent/flink-connector-testing/src/main/java/org/apache/flink/connectors/test/common/utils/TestDataMatchers.java

```diff
@@ -151,11 +156,15 @@ protected boolean matchesSafely(Iterator<T> resultIterator, Description descript
                 final T record = resultIterator.next();
                 if (!matchThenNext(record)) {
                     mismatchDescription = String.format("Unexpected record '%s'", record);
```

Comment:
Apparently this mismatch description is not very helpful. In the very least, I'd add offset and the expected record.